frontend/src/metabase/visualizer/visualizer.slice.ts

```diff
@@ -255,18 +255,21 @@ const visualizerHistoryItemSlice = createSlice({
     builder
       .addCase(setDisplay.fulfilled, (state, action) => {
         const { display, computedSettings } = action.payload;
-        const { columnValuesMapping, columns, settings } =
-          getUpdatedSettingsForDisplay(
-            state.columnValuesMapping,
-            state.columns,
-            computedSettings,
-            state.display,
-            display,
-          );
-
-        state.columnValuesMapping = columnValuesMapping;
-        state.columns = columns;
-        state.settings = settings;
+        const updatedSettings = getUpdatedSettingsForDisplay(
+          state.columnValuesMapping,
+          state.columns,
+          computedSettings,
+          state.display,
+          display,
+        );
+
+        if (updatedSettings) {
+          const { columnValuesMapping, columns, settings } = updatedSettings;
+          state.columnValuesMapping = columnValuesMapping;
+          state.columns = columns;
+          state.settings = settings;
+        }
+
```

Comment:
I think it'd be cleaner if we do the `!display` and `!state.display` checks here, so we don't call `getUpdatedSettingsForDisplay` in the first place, wdyt?

Reply:
Not sure about this, I kinda like the fact that `getUpdatedSettingsForDisplay` accepts anything.